src/main/java/org/opensearch/eval/engine/OpenSearchEngine.java

```diff
@@ -645,8 +645,8 @@ public Map<Integer, Double> getRankAggregatedClickThrough(final int maxRank) thr
                 final List<LongTermsBucket> positionBuckets = positionTerms.lterms().buckets().array();
 
                 for(final LongTermsBucket positionBucket : positionBuckets) {
-                    LOGGER.debug("Inserting client event from position {} with click count {}", positionBucket.key(), (double) positionBucket.docCount());
-                    impressionCounts.put(Integer.valueOf(positionBucket.key()), (double) positionBucket.docCount());
+                    LOGGER.debug("Inserting click event from position {} with click count {}", positionBucket.key(), (double) positionBucket.docCount());
+                    clickCounts.put(Integer.valueOf(positionBucket.key()), (double) positionBucket.docCount());
```

Comment:
This was causing the all zeroes because it was going into the wrong map...

Reply:
nice find